.github/workflows/nightly.yml

```diff
@@ -54,3 +55,38 @@ jobs:
 
       - name: Valgrind Checks
         run: python3 ./src/scripts/ci_build.py --cc=gcc --make-tool=make valgrind-full
+
+  tls_anvil_server_test:
+    name: "TLS-Anvil - Botan Server Test"
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Fetch Botan Repository
+        uses: actions/checkout@v3
+
+      - name: Setup Build Agent
+        uses: ./.github/actions/setup-build-agent
+        with:
+          target: static
+          cache-key: linux-gcc-x86_64-static
```

Comment:
Do we need a separate cache key because we build with `--with-boost`?

Reply:
Oh probably so - maybe best to just use a new unique cache id for each job and let the cache management system sort it out rather than trying to share caches and causing conflicts and cache misses if/when the configuration of different jobs changes over time.

Reply:
Also: if we use a unique target here (eg `anvil`) we can install boost in `setup_gh_actions.sh` as a side effect of setting up the build agent, rather than having an explicit install of boost later on in the yaml

Reply:
Done. (see a589ed2)

Thanks for your feedback!